packages/runtime/src/buckets-utils.tsx

```diff
@@ -0,0 +1,44 @@
+export type Bucket = '' | 'l' | 'v' | 'fw' | 'f' | 'fv' | 'h' | 'a' | 'm';
+
+export const buckets: Bucket[] = ['', 'l', 'v', 'fw', 'f', 'fv', 'h', 'a', 'm'];
+
+// NOTE: If we are adding/removing anything from this list, Please also update the
+// the list in css package. Going forward we might move this common
+// variable in separate package.
+const pseudosMap: { [key: string]: Exclude<Bucket, '' | 'm'> } = {
+  link: 'l',
+  visited: 'v',
+  'focus-within': 'fw',
+  focus: 'f',
+  'focus-visible': 'fv',
+  hover: 'h',
+  active: 'a',
+};
+
+export const getBucket = (sheet: string): Bucket => {
+  if (sheet.charCodeAt(0) === 64) {
```

Comment:
can we add some comments here saying what we're doing

Reply:
Yeah JSDocs and comments are still pending 😅